src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostBuilder.cs

```diff
@@ -177,16 +175,15 @@ public void ConfigureContainer<TBuilder>(IServiceProviderFactory<TBuilder> facto
         public WebAssemblyHost Build()
         {
             // Intentionally overwrite configuration with the one we're creating.
-            var configuration = Configuration.Build();
-            Services.AddSingleton<IConfiguration>(configuration);
+            Services.AddSingleton<WebAssemblyHostConfiguration>(Configuration);
```

Comment:
I think it needs to stay as `Services.AddSingleton<IConfiguration>(configuration)` so that consumers can still receive instances from DI using `@inject IConfiguration MyConfig`.

If we also wanted people to be able to receive it via `@inject WebAssemblyHostConfiguration MyConfig` we would need a second registration, but I don't think there's any use case for that.

Reply:
It is a bit quirky that ASP.NET Core's DI system doesn't resolve requests for subtypes, but does simplify customers' understanding of what's going on (no need for prioritisation rules).

Reply:
> It is a bit quirky that ASP.NET Core's DI system doesn't resolve requests for subtypes, but does simplify customers' understanding of what's going on (no need for prioritisation rules).

True. I've opted to remove it from the dependency injection. Users still have the flexibility to register it themselves within their startup logic but I think it is sensible to not auto-do it here.